src/Components/FinanceDataInput/FinanaceDataInput.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import './FinanceDataInput.scss';
+
+const FinanceDataInput = (prop: any) => {
+    const [value, setValue] = useState('');
+    const [error, setErrorState] = useState (false);
+    const labelErrorMessage = document.getElementById('label');
```

Comment:
issue (blocking): Unfortunately this won't work when there are more than one of these components on the page as they will all share the same ID.

It would be better to conditionally render the error message when there is an error. That way the original label can still be visible. E.g.

```jsx
const FinanceDataInput = () => {

    ...

    return (
        <div className='centered'>
            <div className='group'>
                <input value={value} type={prop.type} onChange={handleChange} id='name' required={prop.required}/>
                <label htmlFor="name" className={error ? 'redLabel' : 'label'} id='label'>{prop.label}</label>
                {error ? "Some error message here." : ""}
                <div className={error ? 'errorBar' : 'bar'}></div>
            </div>
        </div>
    )
}
```